lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -636,32 +653,70 @@ def test_completion_events_with_completion_disabled(self, signal, data):
                 mock_complete.assert_not_called()
             self.assertFalse(BlockCompletion.objects.filter(block_key=block.scope_ids.usage_id).exists())
 
-    @ddt.data(
-        ('complete', {'completion': 0.625}, 0.625),
-        ('progress', {}, 1.0),
-    )
-    @ddt.unpack
     @XBlock.register_temp_plugin(StubCompletableXBlock, identifier='comp')
-    def test_completion_events(self, signal, data, expected_completion):
+    def test_completion_signal_for_completable_xblock(self):
         with completion_waffle.waffle().override(completion_waffle.ENABLE_COMPLETION_TRACKING, True):
             course = CourseFactory.create()
             block = ItemFactory.create(category='comp', parent=course)
             request = self.request_factory.post(
                 '/',
-                data=json.dumps(data),
+                data=json.dumps({'completion': 0.625}),
+                content_type='application/json',
+            )
+            request.user = self.mock_user
+            response = render.handle_xblock_callback(
+                request,
+                unicode(course.id),
+                quote_slashes(unicode(block.scope_ids.usage_id)),
+                'complete',
+                '',
+            )
+            self.assertEqual(response.status_code, 200)
+            completion = BlockCompletion.objects.get(block_key=block.scope_ids.usage_id)
+            self.assertEqual(completion.completion, 0.625)
+
+    @XBlock.register_temp_plugin(StubCompletableXBlock, identifier='comp')
+    def test_progress_signal_ignored_for_completable_xblock(self):
+        with completion_waffle.waffle().override(completion_waffle.ENABLE_COMPLETION_TRACKING, True):
```

Comment:
Minor thing: lines 680-695 and lines 701-716 are identical with the exception of the value of `category` - so it might be worth extracting a little helper method that returns `response`, have the test method use that, and then make assertions.